tracee-ebpf/pkg/metrics/metrics.go

```diff
@@ -0,0 +1,60 @@
+package metrics
+
+import (
+	"github.com/aquasecurity/tracee/tracee-ebpf/tracee"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func RegisterEbpfEngineMetrics(t *tracee.Tracee) error {
+	err := prometheus.Register(prometheus.NewCounterFunc(prometheus.CounterOpts{
+		Namespace: "tracee_ebpf",
+		Name:      "events_total",
+		Help:      "events collected by tracee-ebpf",
+	}, func() float64 { return float64(t.GetStats().EventCount) }))
```

Comment:
You are now exposing prometheus endpoint - and requests can come in parallel.
`Increment()`, correctly, atomically increments the counter. However, read operations from the counter should be done atomically as well - in order to prevent race conditions.

Reply:
Done through counter.Read()